pkg/internal/server/grpc/get_metadata_delta_test.go

```diff
@@ -244,3 +248,389 @@ func TestValidateGetMetadataDeltaRequest(t *testing.T) {
 		})
 	}
 }
+
+func TestConvertToCSIGetMetadataDeltaRequest(t *testing.T) {
```

Comment:
Please add a test that shows that `convertToCSIGetMetadataDeltaRequest` fails if the 2 snapshots come from different volumes - I actually don't see that check in the code at this point!

Reply:
I will raise a separate PR for this.